### ISSUE TITLE
Carthage Fails to resolve dependencies on Carthage Clone

### ISSUE THREAD

**Issue by jeffreybergier** [Carthage/Carthage#421]
Hi, I'm trying to build a project around CarthageKit. I set up a Cartfile as the following:
```
github "Carthage/Carthage" ~> 0.6.4
github "smileyborg/PureLayout" ~> 2.0.5
```
I'm running carthage 0.6.4 and the output is:
```
jeffberg@PalladumVM ~/D/S/Cartwheel> carthage update
*** Fetching Carthage
*** Fetching PureLayout
*** Fetching ReactiveCocoa
*** Fetching LlamaKit
*** Fetching ReactiveTask
*** Fetching Argo
*** Fetching LlamaKit
*** Fetching LlamaKit
Could not pick a version for github "ReactiveCocoa/ReactiveCocoa", due to mutually incompatible requirements:
	"swift-development"
	"v3.0-alpha.3"
jeffberg@PalladumVM ~/D/S/Cartwheel> carthage version
0.6.4
```

Using PureLayout on its own works just fine. Removing the version requirement for Carthage lowers the version it downloads to 0.3 and the build fails on some of the dependencies.

I also cloned the Carthage repo to my disk and ran carthage update and got the same error as above.

Any help would be greatly appreciated.
-Jeff

**Comment by jspahrsummers** [comment]
Sorry about that. This is an issue because Carthage is currently [pinned to a branch of ReactiveCocoa](https://github.com/Carthage/Carthage/blob/fb67bc83840ca886c5b4a7fac069f08fba147431/Cartfile#L1), except that the branch has introduced breaking changes since Carthage itself was last updated.

I have a work-in-progress pull request to [upgrade Carthage's dependencies](https://github.com/Carthage/Carthage/pull/368), but I've encountered some blockers in Swift 1.2 that we need to find workarounds for. Until that lands, Carthage won't be able to use the latest `swift-development` branch from ReactiveCocoa.

You can work around this issue by constraining ReactiveCocoa or ReactiveTask to [the exact version](https://github.com/Carthage/Carthage/blob/fb67bc83840ca886c5b4a7fac069f08fba147431/Cartfile.resolved#L8-L9) that Carthage is currently using. Sorry about the trouble!

**Comment by jeffreybergier** [comment]
Thanks for the response. I'll give it a go tomorrow. After seeing your talk at realm, I had an idea for a simple GUI. It will be my first AppKit app so it's two giant adventures in one :)

**Comment by jspahrsummers** [comment]
@jeffreybergier Awesome! Let us know how it goes. :zap:

**Comment by jeffreybergier** [comment]
Thanks for your help. I tried with all my might to make Carthage build with Carthage and nothing was doing it. I was able to clone this repo and add all the submodules to a workspace and then use CarthageKit in the new app project. It doesn't work with Xcode 6.3 though. I can't wait for the swift 1.2 update. Thanks again for your help. I also filed enhancement related issue #422 
![menubar_and_cartwheel_xcodeproj](https://cloud.githubusercontent.com/assets/7349616/7104347/a7fcb3c0-e08e-11e4-88e4-8402968014a3.png)
